### PR TITLE
Fix links in documentation

### LINKED ISSUE #1422: [Issue] Broken Relative paths
These files / heading references no longer exist on the repo.
Need to reference correct files

| File Full Path | Issues |
|--------|--------|
| `./docs/productionizing.md` | 1. `../README.md#enabling-authentication` |
| `./docs/deploy_features.md` | 1. `../README.md#deploying` |
| `./docs/login_and_acl.md` | 1. `./app/backend/core/authentication.py` <br/> 2. `./scripts/adlsgen2setup.ps1` <br/> 3. `./scripts/sampleacls.json` <br/> 4. `./scripts/sampleacls.json` <br/> 5. `./scripts/sampleacls.json` <br/> 6. `./scripts/sampleacls.json` <br/> 7. `./scripts/prepdocs.py` <br/> 8. `./scripts/manageacl.ps1` <br/> 9. `./scripts/manageacl.ps1` <br/> 10. `./scripts/manageacl.sh` |
| `./docs/azd.md` | 1. `../README.md#deploying-from-scratch` |
| `./docs/deploy_existing.md` | 1. `../README.md#deploying` |
| `./docs/deploy_lowcost.md` | 1. `../README.md#deploying` <br/> 2. `../README.md#existing-azure-ai-search-resource` <br/> 3. `../README.md#deploying-from-scratch` |
| `./docs/localdev.md` | 1. `../README.md#azure-deployment` |
| `./.github/PULL_REQUEST_TEMPLATE.md` | 1. `../CONTRIBUTING.md#submit-pr` |

_Originally posted by @github-actions[bot] in https://github.com/Azure-Samples/azure-search-openai-demo/issues/1386#issuecomment-1992538667_

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

Fixes #1422 

This PR fixes links in docs, preferring absolute URLs over relative URLs which are more likely to break if docs get moved around. The absolute URLs seem to work, from my tests.

Also updated a few docs sections while I was in them. :)

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](../CONTRIBUTING.md#submit-pr) for more details.

N/A

### CODE REVIEW

File: docs/productionizing.md [hunk 1]
```diff
@@ -23,7 +24,7 @@ If the maximum TPM isn't enough for your expected load, you have a few options:
...
 
-* If you are consistently going over the TPM, then consider implementing a load balancer between OpenAI instances. Most developers implement that using Azure API Management following [this blog post](https://www.raffertyuy.com/raztype/azure-openai-load-balancing/) or [this repository](https://github.com/andredewes/apim-aoai-smart-loadbalancing). Another approach is to use [LiteLLM's load balancer](https://docs.litellm.ai/docs/providers/azure#azure-api-load-balancing) with Azure Cache for Redis.
+* If you are consistently going over the TPM, then consider implementing a load balancer between OpenAI instances. Most developers implement that using Azure API Management using [the openai-apim-lb repo](https://github.com/Azure-Samples/openai-apim-lb) or with Azure Container Apps using [the openai-aca-lb repo](https://github.com/Azure-Samples/openai-aca-lb). Another approach is to use [LiteLLM's load balancer](https://docs.litellm.ai/docs/providers/azure#azure-api-load-balancing) with Azure Cache for Redis.
```
Comment (pamelafox) [inline docs/productionizing.md:27]: New load balancer links

File: docs/productionizing.md [hunk 2]
```diff
@@ -99,3 +100,7 @@ Here's an example loadtest for 50 users and a spawn rate of 1 per second:
 ![Screenshot of Locust charts showing 5 requests per second](images/screenshot_locust.png)
 
 After each test, check the local or App Service logs to see if there are any errors.
+
+## Evaluation
+
+Before you make your chat app available to users, you'll want to rigorously evaluate the answer quality. You can use tools in [the AI RAG Chat evaluator](https://github.com/Azure-Samples/ai-rag-chat-evaluator) repository to run evaluations, review results, and compare answers across runs.
```
Comment (pamelafox) [inline docs/productionizing.md:106]: New section to encourage evaluation (this section is also in customizing doc)

**Comment by pamelafox** [comment]
cc @john0isaac

**Comment by john0isaac** [comment]
The workflow is not passing it's skipping the URLs now.
To catch the URL by the workflow it needs to start with ./ or ../

Also, the relative reference is needed to be able to evaluate it relatively.

**Comment by john0isaac** [comment]
for example to fix the comments posted by the workflow you needed to add an extra `.` in front of the /app
change ./app/backend/core/authentication.py to ../
removing the `.` dots makes it skip it.

**Review by tonybaloney — APPROVED** [review]

**Comment by pamelafox** [comment]
Changed to relative paths, and checker is still passing.

**Pull request merged**